### PR TITLE
Oe pom entity default

### PR DESCRIPTION
**Pull request by OxanaE**
https://trello.com/c/Ew7XqiIn/427-pom-entitydefault-test
Refactored 2 tests in class out of 3 tests  to use POM except the part that uses BaseViewPage and embedded values on BaseEditPage.

### CODE REVIEW

File: src/test/java/EntityDefaultTest.java [hunk 1]
```diff
@@ -161,6 +149,15 @@ private void selectFromRecordMenu (WebDriver driver, By byFunction) {
...
 
+    private void assertRecordValues(List<String> actual_values, String[] expected_values) {
+        Assert.assertEquals(actual_values.size(), expected_values.length);
+        for (int i = 0; i < expected_values.length; i++) {
+            if (expected_values[i] != null) {
+                Assert.assertEquals(actual_values.get(i), expected_values[i]);
+            }
+        }
+    }
+
```
Comment (Stan575) [inline src/test/java/EntityDefaultTest.java:160]: assertion can be done in test using actual and expected data sets of the same type

File: src/test/java/EntityDefaultTest.java [hunk 2]
```diff
@@ -176,18 +173,15 @@ public void checkDefaultValuesAndUpdate() {
...
-        assertAndReplace(driver, BY_TEXT, defaultValues.fieldText, changedDefaultValues.fieldText, false);
-        assertAndReplace(driver, BY_INT, defaultValues.fieldInt, changedDefaultValues.fieldInt, false);
-        assertAndReplace(driver, BY_DECIMAL, defaultValues.fieldDecimal, changedDefaultValues.fieldDecimal, false);
-        assertAndReplace(driver, BY_DATE, defaultValues.fieldDate, changedDefaultValues.fieldDate, false);
-        assertAndReplace(driver, BY_DATETIME, defaultValues.fieldDateTime, changedDefaultValues.fieldDateTime, false);
-        assertAndReplaceFieldUser(driver, defaultValues.fieldUser, changedDefaultValues.fieldUser, BY_USER, BY_DROPDOWN);
+        defaultEditPage.sendKeys(changedDefaultValues.fieldString, changedDefaultValues.fieldText, changedDefaultValues.fieldInt,
+                changedDefaultValues.fieldDecimal, changedDefaultValues.fieldDate, changedDefaultValues.fieldDateTime, changedDefaultValues.fieldUser);
 
         WebElement createEmbedD = driver.findElement(By.xpath("//button[@data-table_id='11']"));
```
Comment (Stan575) [inline src/test/java/EntityDefaultTest.java:186]: move all interactions with browser to a page class

**Review by Stan575 — CHANGES_REQUESTED** [review]

File: src/test/java/model/DefaultEditPage.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    @Override
+    protected DefaultPage createPage() {
+        return new DefaultPage(getDriver());
+    }
+
+    private void sendKeys(WebElement element, String newValue){
+        element.clear();
+        element.sendKeys(newValue);
+        element.sendKeys("\t");
```
[the remark this answers is not in the document]
Comment (OxanaE) [reply src/test/java/model/DefaultEditPage.java:77]: sends tab. without it one of the fields don't work

File: src/test/java/EntityDefaultTest.java (hunk 2 above)
Comment (OxanaE) [reply src/test/java/EntityDefaultTest.java:186]: this part is not yet refactored

File: src/test/java/model/BaseTablePage.java [hunk 4]
```diff
@@ -55,7 +55,11 @@ public WebElement getRowEntityIcon(int rowNumber) {
     }
 
     public List<String> getRow(int rowNumber) {
-        return getRows().get(rowNumber).findElements(By.xpath("//td/a/div")).stream()
+        return getRow(rowNumber, "//td/a/div");
+    }
+
+    public List<String> getRow(int rowNumber, String xpath) {
+        return getRows().get(rowNumber).findElements(By.xpath(xpath)).stream()
```
[the remark this answers is not in the document]
Comment (OxanaE) [reply src/test/java/model/BaseTablePage.java:62]: in my code old functionality and method signature is preserved so none of the existing code gets affected. the new code adds flexibility without code duplication. see lines 57 .. 59

`
    public List<String> getRow(int rowNumber) {
        return getRow(rowNumber, "//td/a/div");
    }

    public List<String> getRow(int rowNumber, String xpath) {
        return getRows().get(rowNumber).findElements(By.xpath(xpath)).stream()
                .map(WebElement::getText).collect(Collectors.toList());
    }
`

File: src/test/java/EntityDefaultTest.java (hunk 1 above)
Comment (OxanaE) [reply src/test/java/EntityDefaultTest.java:160]: Refactored to use assert of equal types where it's possible. There's still one test that needs refactoring later.

**Comment by OxanaE** [comment]
One of the tests (the first one) uses a DefaultViewPage pom which is not ready yet. I'm waiting for someone more knowledgeable to do it. This is the reason I didn't refactor this test yet.

**Review by GalinaRuban — APPROVED** [review]

**Review by SergeiDemyanenko — APPROVED** [review]

**Review by Stan575 — APPROVED** [review]

**Pull request merged**